llvm/lib/Target/DirectX/DirectX.h

```diff
@@ -28,6 +28,12 @@ void initializeDXILPrepareModulePass(PassRegistry &);
 /// Pass to convert modules into DXIL-compatable modules
 ModulePass *createDXILPrepareModulePass();
 
+/// Initializer for DXIL strength reduce
+void initializeDXILStrengthReducePass(PassRegistry &);
+
+/// Pass to reduce strength during lowering into DXIL-compatable modules
+ModulePass *createDXILStrengthReducePass();
```

Comment:
Do we need DXILStrengthReducePass for add these new intrinsics?

Reply:
This pass is introduced to rewrite intrinsics such as `llvm.abs.*` (integer abs) with no corresponding DXIL intrinsic (AFAIK). For example, `llvm.abs` is rewritten in this pass as a combination of `sub` (LLVM instruction) and `smax` (LLVM intrinsic) - as done in DXC.

Reply:
It looks like you are only implementing the strenght reduce for `llvm.abs.*` and `fneg` here, but are adding many other intrinsics.

I think that in general we should strive to make the PRs in the smallest self-contained way possible. This makes it easier to review (and people are also usually more willing to jump in and review smaller changes).

I see this as three logical PRs here

1. Add the strength reduce pass and handle the `fneg` instruction.
2. Extend the strength reduce pass to also lower the `llvm.abs` intrinsics
3. Add the other intrinsics that do not depend on the strenght reduce pass (e.g. `llvm.cos`).

The first PR is basically a refactoring (do we have exisiting test coverage for `fneg`)? And separating the other PRs makes them easier to review and give constructive feedback.